### PR TITLE
Publish IJ plugin snapshots

### PR DESCRIPTION
**Pull request by BoD**
Publish the IJ plugin to the Maven central snapshots repository.

This should make it available when adding this custom repository in the plugins settings: `https://s01.oss.sonatype.org/content/repositories/snapshots/com/apollographql/apollo-intellij-plugin/<version>-SNAPSHOT/apollo-intellij-plugin-<version>-SNAPSHOT.xml`.

### CODE REVIEW

File: intellij-plugin/build.gradle.kts [hunk 1]
```diff
@@ -148,3 +149,47 @@ tasks.register("downloadMockJdk") {
 tasks.named("test").configure {
   dependsOn("downloadMockJdk")
 }
+
+tasks.register("generateUpdatePluginsXml") {
+  val filePath = "build/publications/updatePlugins.xml"
+  val pluginName = properties("pluginName")
+  val version = properties("VERSION_NAME")
```
Comment (martinbonnin) [inline intellij-plugin/build.gradle.kts:156]: You need to mark this as input or else the task might be always up-to-date

File: intellij-plugin/build.gradle.kts [hunk 2]
```diff
@@ -148,3 +149,47 @@ tasks.register("downloadMockJdk") {
 tasks.named("test").configure {
   dependsOn("downloadMockJdk")
 }
+
+tasks.register("generateUpdatePluginsXml") {
```
Comment (martinbonnin) [inline intellij-plugin/build.gradle.kts:153]: Are tasks created like this cacheable by default?

File: intellij-plugin/build.gradle.kts [hunk 3]
```diff
@@ -148,3 +149,47 @@ tasks.register("downloadMockJdk") {
...
+        password = System.getenv("SONATYPE_NEXUS_PASSWORD")
+      }
+    }
+  }
+
+  publications {
+    create<MavenPublication>("default") {
+      artifactId = properties("pluginName")
+      artifact(tasks.named("signPlugin"))
+      artifact(tasks.named("generateUpdatePluginsXml"))
```
Comment (martinbonnin) [inline intellij-plugin/build.gradle.kts:192]: Shouldn't this publish the plugin .jar as well?

File: intellij-plugin/build.gradle.kts (hunk 2 above)
Comment (BoD) [reply intellij-plugin/build.gradle.kts:153]: Woops didn't think about the cache, good question, I'll 👀

File: intellij-plugin/build.gradle.kts (hunk 3 above)
Comment (BoD) [reply intellij-plugin/build.gradle.kts:192]: It looks like this only publishes the .zip (the output of `signPlugin`), which is all that we need?

Comment (martinbonnin) [reply intellij-plugin/build.gradle.kts:192]: Looks like it 👍

File: intellij-plugin/build.gradle.kts (hunk 2 above)
Comment (BoD) [reply intellij-plugin/build.gradle.kts:154]: Nope, you must [ask for it explicitly](https://docs.gradle.org/current/userguide/build_cache.html#using_the_runtime_api), TIL!

**Review by martinbonnin — APPROVED** [review]

**Pull request merged**